### PR TITLE
[7.17] Update Azure SDK to version 12.20.1

### PR DESCRIPTION
**Pull request by fcofdez**

### CODE REVIEW

File: build-tools-internal/version.properties [hunk 1]
```diff
@@ -7,7 +7,7 @@ bundled_jdk = 19.0.1+10@afdd2e245b014143b62ccb916125e3ce
 # optional dependencies
 spatial4j         = 0.7
 jts               = 1.15.0
-jackson           = 2.10.4
```
Comment (fcofdez) [inline build-tools-internal/version.properties:10]: This upgrade is required for the newer SDK versions to work

**Comment by fcofdez** [comment]
Adding @rjernst and @ChrisHegarty as reviewers due to the broader Jackson upgrade.

**Review by arteam — APPROVED** [review]
LGTM! Thanks Francisco!

**Review by rjernst — APPROVED** [review]
LGTM. Glad to see Jackson (and Azure) finally upgradeable.

**Review by ChrisHegarty — APPROVED** [review]
LGTM

**Comment by fcofdez** [comment]
@elasticmachine update branch

**Comment by fcofdez** [comment]
Thanks all!

**Pull request merged**